openpype/hosts/traypublisher/api/editorial.py

```diff
@@ -0,0 +1,327 @@
+import re
+from copy import deepcopy
+
+from openpype.client import get_asset_by_id
+from openpype.pipeline.create import CreatorError
+
+
+class ShotMetadataSolver:
+    """ Solving hierarchical metadata
+
+    Used during editorial publishing. Works with imput
+    clip name and settings defining python formatable
+    template. Settings also define searching patterns
+    and its token keys used for formating in templates.
+    """
+
+    NO_DECOR_PATERN = re.compile(r"\{([a-z]*?)\}")
+
+    # presets
+    clip_name_tokenizer = None
+    shot_rename = True
+    shot_hierarchy = None
+    shot_add_tasks = None
+
+    def __init__(
+        self,
+        clip_name_tokenizer,
+        shot_rename,
+        shot_hierarchy,
+        shot_add_tasks,
+        logger
+    ):
+        self.clip_name_tokenizer = clip_name_tokenizer
+        self.shot_rename = shot_rename
+        self.shot_hierarchy = shot_hierarchy
+        self.shot_add_tasks = shot_add_tasks
+        self.log = logger
+
+    def _rename_template(self, data):
+        """Shot renaming function
+
+        Args:
+            data (dict): formating data
+
+        Raises:
+            CreatorError: If missing keys
+
+        Returns:
+            str: formated new name
+        """
+        shot_rename_template = self.shot_rename[
+            "shot_rename_template"]
+        try:
+            # format to new shot name
+            return shot_rename_template.format(**data)
+        except KeyError as _E:
+            raise CreatorError((
+                "Make sure all keys are correct in settings: \n\n"
+                f"From template string {shot_rename_template} > "
+                f"`{_E}` has no equivalent in \n"
+                f"{list(data.keys())} input formating keys!"
+            ))
+
+    def _generate_tokens(self, clip_name, source_data):
+        """Token generator
+
+        Settings defines token pairs key and regex expression.
+
+        Args:
+            clip_name (str): name of clip in editorial
+            source_data (dict): data for formating
+
+        Raises:
+            CreatorError: if missing key
+
+        Returns:
+            dict: updated source_data
+        """
+        output_data = deepcopy(source_data["anatomy_data"])
+        output_data["clip_name"] = clip_name
+
+        if not self.clip_name_tokenizer:
+            return output_data
+
+        parent_name = source_data["selected_asset_doc"]["name"]
+
+        search_text = parent_name + clip_name
+
+        for token_key, pattern in self.clip_name_tokenizer.items():
+            p = re.compile(pattern)
+            match = p.findall(search_text)
+            if not match:
+                raise CreatorError((
+                    "Make sure regex expression is correct: \n\n"
+                    f"From settings '{token_key}' key "
+                    f"with '{pattern}' expression, \n"
+                    f"is not able to find anything in '{search_text}'!"
+                ))
```

Comment:
```suggestion
                raise CreatorError((
                    "Make sure regex expression works with your data: \n\n"
                    f"Regex pattern '{pattern}' in your settings"
                    f"can't find any match in your clip name '{search_text}'!\n"
                ))
```

This error has a bit complicated word order, which makes it quite hard to decipher. I suggest something more easily readable like this.